tests/test_2772_parquet_extn_array_metadata.py

```diff
@@ -137,6 +138,22 @@ def test_array_conversions(akarray, as_dict):
     rt_array = ak.from_arrow(as_extn, highlevel=True)
     assert to_list(rt_array) == to_list(akarray)
 
+    # Deeper test of types
+    akarray_high = ak.Array(akarray)
+    if akarray_high.type.content.parameters.get("__categorical__", False) == as_dict:
+        # as_dict is supposed to go hand-in-hand with __categorical__: True, and if it
+        #  does not, we do not round-trip perfectly. So only test when this is set correctly.
+        assert rt_array.type == akarray_high.type
+
+        ak_type_str_orig = io.StringIO()
+        ak_type_str_rtrp = io.StringIO()
+        akarray_high.type.show(stream=ak_type_str_orig)
+        rt_array.type.show(stream=ak_type_str_rtrp)
+        if ak_type_str_orig.getvalue() != ak_type_str_rtrp.getvalue():
+            print("  Original type:", ak_type_str_orig.getvalue())
+            print("  Rnd-trip type:", ak_type_str_rtrp.getvalue())
+        assert ak_type_str_orig.getvalue() == ak_type_str_rtrp.getvalue()
```

Comment:
This depends on us not changing the way that types are represented as strings. Well, it depends on not introducing any differences between the original type and the round-trip type, which is unlikely. There must be a better way to compare types with your chosen granularity (e.g. preserving the distinction between `X` and `Categorical[X]`).

Reply:
I think this ends up testing the parameters for the nested types, which doesn't happen automatically. I was stealing the approach from [tests/test_1440_start_v2_to_parquet.py](https://github.com/scikit-hep/awkward/blob/e9dd2694cf5c6ccb1717358fe926816d8ecbfb9c/tests/test_1440_start_v2_to_parquet.py#L32-L39).